### PR TITLE
Remove ansible_env and normalize defaulting vars

### LINKED ISSUE #3: Install script misuses ansible_env for many variables that are NOT in ansible_env
@kholdaway commented on [Fri May 17 2019](https://github.com/quipucords/quipucords/issues/1924)

## Bug Description:
Install script misuses ansible_env for many variables that are NOT in ansible_env

These luckily work due to defaults.  Code should be fixed.

### LINKED ISSUE #6: Install script open port flag fails when set to true/false/anything
@kholdaway commented on [Fri May 17 2019](https://github.com/quipucords/quipucords/issues/1923)

## Bug Description:
When I set the open port flag on the install script, the script bombs out

### Expected behavior:
install.sh -e open_port=false

### LINKED ISSUE #7: Install script does not recognize install_offline=false
@abaiken commented on [Fri May 17 2019](https://github.com/quipucords/quipucords/issues/1922)

## Bug Description:
When i run `./install.sh -e install_offline=false` it still checks to see if the files necessary for offline install exist even though I specified it as false. 

### Expected behavior:
If a user sets install_offline=false we should not fail if the files needed for offline install do not exist.

### Actual behavior:
```
[vagrant@vrhel6 vagrant_data]$ ./install.sh -e install_offline=false
Checking if required files exist for an offline installation.
packages//quipucords.1.0.0.tar.gz is required for an offline installation.
```

### Steps to reproduce:
Install qpc on a vagrant machine with the `install_offline=false` argument. Make sure that you do not have the tar.gz in your packages directory.

___

## Acceptance Criteria:

- [ ] Verify that install_offline can be set to false & work as expected

___

### Version Information:
CLI version, server version or git commit information.

### Environment information:
Provide detailed information on the *scanning* and *target* systems (if applicable); see example below.

Type            | Operating System | OS Version | Python Version | Virtualization
--------------- | ---------------- | ---------- | -------------- | --------------
Scanning System | RHEL             | 7.4        | 3.6            | Bare Metal
Target System   | RHEL             | 5.9        | 2.4.6          | RHEV

### PR DESCRIPTION
**Pull request by kholdaway**
Closes #3 
Closes #6 
Closes #7

### CODE REVIEW

File: install/roles/prep/tasks/main.yml [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
-    - install_offline == 'true'
-    
-- name: Set install_offline flag to false it was not defined or was not true
-  set_fact:
-    install_flag_install_offline: false 
-  when: 
-    - install_flag_install_offline is not defined
+- name: Validate install_server is set to true or false
+  fail:
+    msg: "{{ install_server}} is invalid value for install_server.  Must be equal to true or false"
```
Comment (Dostonbek1) [inline install/roles/prep/tasks/main.yml]: Possibly a space needed after `install_server` and before closing curly braces.

File: install/roles/prep/tasks/main.yml [hunk 2]
```diff
@@ -1,23 +1,32 @@
...
+  when:
+    - not install_server|lower == 'true'
+    - not install_server|lower == 'false'
+
+- name: Validate install_cli is set to true or false
+  fail:
+    msg: "{{ install_cli}} is invalid value for install_cli.  Must be equal to true or false"
```
Comment (Dostonbek1) [inline install/roles/prep/tasks/main.yml]: Space between `install_cli` and closing curly braces.

File: install/roles/prep/tasks/main.yml [hunk 3]
```diff
@@ -1,23 +1,32 @@
...
+  when:
+    - not install_cli|lower == 'true'
+    - not install_cli|lower == 'false'
+
+- name: Validate install_offline is set to true or false
+  fail:
+    msg: "{{ install_offline}} is invalid value for install_offline.  Must be equal to true or false"
```
Comment (Dostonbek1) [inline install/roles/prep/tasks/main.yml]: Space between `install_offline` and closing curly braces.

File: install/roles/qpc_server/tasks/main.yml [hunk 4]
```diff
@@ -5,20 +5,19 @@
...
-    server_port: "{{ ansible_env['SERVER_PORT'] | default('9443') }}"
-  when: server_port is not defined
-
-- name: Set whether installer should open port on host machine
-  set_fact:
-    open_port: "{{ ansible_env['OPEN_PORT'] | bool | default(true) }}"
-  when: open_port is defined
+- name: Validate use_supervisord is set to true or false
+  fail:
+    msg: "{{ use_supervisord}} is invalid value for use_supervisord.  Must be equal to true or false"
```
Comment (Dostonbek1) [inline install/roles/qpc_server/tasks/main.yml]: Space between `user_supervisord` and closing curly braces.

File: install/roles/qpc_server/tasks/main.yml [hunk 5]
```diff
@@ -5,20 +5,19 @@
...
+    - not use_supervisord|lower == 'true'
+    - not use_supervisord|lower == 'false'
 
-- name: Defaulting open port setting to true
-  set_fact:
-    open_port: true
-  when: open_port is not defined
+- name: Validate open_port is set to true or false
+  fail:
+    msg: "{{ open_port}} is invalid value for open_port.  Must be equal to true or false"
```
Comment (Dostonbek1) [inline install/roles/qpc_server/tasks/main.yml]: Space between `open_port` and closing curly braces.

File: install/roles/qpc_cli/tasks/main.yml [hunk 6]
```diff
@@ -59,4 +59,4 @@
 - name: Configure QPC to talk to server
   shell: qpc server config --host 127.0.0.1 --port "{{ server_port }}"
   when:
-    - install_server
+    - install_server|lower == 'true'
```
Comment (myersCody) [inline install/roles/qpc_cli/tasks/main.yml]: Do we need this `when` condition anymore? I think it gets covered in the `qpc_playbook.yml`.

Comment (kholdaway) [reply install/roles/qpc_cli/tasks/main.yml]: This check is not covered by the playbook level when.  If you choose not to install the server, you should not run that line.

File: install/roles/prep/tasks/main.yml (hunk 1 above)
Comment (kholdaway) [reply install/roles/prep/tasks/main.yml]: Fixed

File: install/roles/prep/tasks/main.yml (hunk 2 above)
Comment (kholdaway) [reply install/roles/prep/tasks/main.yml]: Fixed

File: install/roles/prep/tasks/main.yml (hunk 3 above)
Comment (kholdaway) [reply install/roles/prep/tasks/main.yml]: Fixed

File: install/roles/qpc_server/tasks/main.yml (hunk 4 above)
Comment (kholdaway) [reply install/roles/qpc_server/tasks/main.yml]: Fixed

File: install/roles/qpc_server/tasks/main.yml (hunk 5 above)
Comment (kholdaway) [reply install/roles/qpc_server/tasks/main.yml]: Fixed

File: install/roles/qpc_cli/tasks/main.yml (hunk 6 above)
Comment (myersCody) [reply install/roles/qpc_cli/tasks/main.yml]: Ohhhh good point. I thought I was looking in `qpc_server` 😬

**Pull request merged**